Goal: Be up front on the origin of our su.
Fixes: #244297

Status wrt upstream: It must be forwarded upstream.
                     Upstream also uses some bits from the GNU shellutils.

Index: shadow-4.0.18.1/src/su.c
===================================================================
--- shadow-4.0.18.1.orig/src/su.c	2006-09-17 12:17:53.099998487 +0200
+++ shadow-4.0.18.1/src/su.c	2006-09-17 12:17:56.308023993 +0200
@@ -26,6 +26,24 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+/* Some parts substantially derived from an ancestor of: */
+/* su for GNU.  Run a shell with substitute user and group IDs.
+   Copyright (C) 1992-2003 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
 
 #include <config.h>
 
